### ISSUE TITLE
Multiple annotation - a bug in saving?

### ISSUE THREAD

**Issue by ChiaraPalladino** [pelagios/recogito2#297]
Not sure about this - I noticed that some of the multiple / multi-layered annotations I tried to take today and yesterday are not saved. When refreshing the page after automatic saving, I can't see them anymore. They are either two- or three-level annotations.

**Comment by rsimon** [comment]
Hm. Just tried this out, layering three annotations on top of each other (starting with the "longest" one, then creating smaller ones inside; and vice versa). That seemed to work just fine on Chrome and Firefox (Windows). So probably not caused by levels themselves, but some other combination of circumstances. Do have any more info that would help to reproduce this?

**Comment by eltonteb** [comment]
Multi-layered annotations worked for me too, though only after Rainer reminded me that I needed to hit "ok" each time.

**Comment by ChiaraPalladino** [comment]
i am using Chrome as well and it really messes up with my layers. I am trying to have a sequence of layered annotations that goes like "placename" + "definition" as it appears in the source + "definition2", each one being a single annotation and then part of a layered annotation. After refreshing the page, generally one of the layers is gone. 

How many annotations do you guys have? I passed 500, and noticed that this started to happen only when the number becomes quite high.

**Comment by rsimon** [comment]
@ChiaraPalladino can you share the doc with me (and point me to an example place where I can try this out)?

**Comment by ChiaraPalladino** [comment]
I hesitated to do that because it's in Greek :) 
Anyway I shared it. A couple of very easy instances are at the third paragraph, Σερβωνίδος λίμνης and Φᾶσιν ποταμόν: I tried to annotate the following layers: "Σερβωνίδος", "Σερβωνίδος λίμνης"; and "Φᾶσιν", "Φᾶσιν ποταμόν". As you can see, there is only one layer saved - strangely "Σερβωνίδος" in the first case, and "Φᾶσιν ποταμόν" in the second.

**Comment by rsimon** [comment]
Ok, thanks. I played around a bit. Can't say for sure what it is yet. But I can reproduce it. My guess is that the annotations _are_ in fact getting stored as they should. It's the UI code that renders the highlights that seems to fail under certain conditions (yet to be figured out).

I'm on it!

**Comment by rsimon** [comment]
Hi @ChiaraPalladino,

I think I caught it & just added a fix. As suspected, annotations did get stored. It was the highlight rendering that would fail when annotations shared the same start location. Can you check if this fixes the issue on your end?

(A warning: you may have produced a few duplicate annotations, which will now re-appear. Let me know if you need help fixing this!)

**Comment by ChiaraPalladino** [comment]
Thanks, I think it's working now - I'll let you know if I see other bugs. 

Regarding this - it would be useful to be able to see how many annotation layers are in place. I don't know how this could be graphically possible, but it would be helpful.

**Comment by rsimon** [comment]
I agree. Ultimately, my plan was to not allow "perfect overlaps" at all (i.e. an annotation that starts and ends at exactly the same characters). Rather, Recogito would open the existing annotation, and then you'd be able to add to this one. 

This would IMO alleviate the issue of layers a bit. For many layers of annotations that overlap partly, I agree there should be better visual representation. I'll think about some solutions. I'll create a separate issue dedicated to this. If the inital bug looks fixed on your end, feel free to close this one.

**Comment by ChiaraPalladino** [comment]
Seems to be fine.
